travis-ci/packer-templates-mac
# Add Packer build templates for Linux vanilla VMs

## What is the problem that this PR is trying to fix?

Building a new Ubuntu vanilla VM for the Mac infra was not super fun, and required several manual steps in vSphere. We were also using the cloudimg versions of the Ubuntu images, which were not ideal for our use case. We also may in the future want to explore making more of our infrastructure immutable using Packer images, so we need some sort of automation in place for that.

## What approach did you choose and why?

I added a new Packer template for the Ubuntu Trusty vanilla VM that we currently use. It uses the `vsphere-iso` builder to run the Ubuntu installer in an automated fashion. Then it provisions the VM with some Ansible playbooks that do the equivalent to what our wjxb-bootstrap script did. One nice thing about this is it gets to reuse the same Ansible role I created to keep the users on the existing Linux VMs up-to-date!

## How can you test this?

First, update your travis-keychain repo!

Build a new Ubuntu VM with `rake ubuntu14.04`. It should show up in the Vanilla VMs folder in both pods when complete.

Also try running `rake update_users` which will likely not change anything but should ensure that the current Linux VMs have the right users.

## What feedback would you like, if any?

Any!
